ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -133,17 +132,18 @@ connections PeerSelectionActions{peerStateActions = PeerStateActions {monitorPee
                                $ (Map.keysSet demotedToCold)
 
         in assert
-            (let establishedPeersSet1 = Map.keysSet establishedPeers1
+            (let establishedPeersSet1 = Map.keysSet (EstablishedPeers.toMap establishedPeers1)
```

Comment:
Yes I agree, we should have `keysSet`.

Reply:
I suggest we have this interface:
```
allPeers :: Ord peeraddr
         => EstablishedPeers peeraddr peerconn
         -> Map peeraddr peerconn
allPeers EstablishedPeers { establishedReady, nextActivateTimes } =
       establishedReady
    <> Map.fromList [ (peeraddr, peerconn)
                    | (peeraddr, _, peerconn) <- PSQ.toList nextActivateTimes ]


allKeysSet :: Ord peeraddr
           => EstablishedPeers peeraddr peerconn
           -> Set peeraddr
allKeysSet = Map.keysSet . establishedStatus


keysSet :: Ord peeraddr
        => EstablishedPeers peeraddr peerconn
        -> Set peeraddr
keysSet = Map.keysSet . establishedReady
```
This module is imported qualified, so at call sites it will be `EstablishedPeers.keysSet` or `EstablishedPeers.allPeers`.

```